File: src/main/java/com/examportal/dto/AnswerDTO.java
package com.examportal.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AnswerDTO {
    
    @NotNull(message = "Question ID must be provided")
    private Long questionId;
    
    @Pattern(regexp = "[ABCD]", message = "Selected option must be A, B, C, or D")
    private String selectedOption;
}

File: Dotnet3/ExamPortal/DTOs/AnswerDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ExamPortal.DTOs
{
    public class AnswerDTO
    {
        [Required(ErrorMessage = "Question ID must be provided")]
        public long QuestionId { get; set; }
        
        [Required(ErrorMessage = "Selected option must be provided")]
        [RegularExpression("[ABCD]", ErrorMessage = "Selected option must be A, B, C, or D")]
        public string SelectedOption { get; set; } = string.Empty;
    }
}
